unitylibs/core/workflow/workflow-upload/action-binder.js

```diff
@@ -0,0 +1,354 @@
+    unityConfig.psEndPoint = {
+      assetUpload: `${unityConfig.apiEndPoint}/asset`,
+      acmpCheck: `${unityConfig.apiEndPoint}/asset/finalize`,
+    };
+    return unityConfig;
+  }
+
+  async handlePreloads() {
+    const parr = [];
+    if (this.workflowCfg.targetCfg.showSplashScreen) {
+      parr.push(
+        `${getUnityLibs()}/core/styles/splash-screen.css`,
+      );
+    }
+    await priorityLoad(parr);
+  }
+
+  async cancelUploadOperation() {
+    try {
+      document.querySelector('a.con-button[href*="#_cancel"]').setAttribute('daa-ll', 'cancel');
+      const { default: TransitionScreen } = await import(`${getUnityLibs()}/scripts/transition-screen.js`);
+      this.transitionScreen = new TransitionScreen(this.transitionScreen.splashScreenEl, this.initActionListeners, this.LOADER_LIMIT, this.workflowCfg);
+      await this.transitionScreen.showSplashScreen();
+      const e = new Error('Operation termination requested.');
+      const cancelPromise = Promise.reject(e);
+      this.promiseStack.unshift(cancelPromise);
+    } catch (error) {
+      await this.transitionScreen?.showSplashScreen();
+      throw error;
+    }
+  }
+
+  extractFiles(e) {
+    const files = [];
+    if (e.dataTransfer?.items) {
+      [...e.dataTransfer.items].forEach((item) => {
+        if (item.kind === 'file') files.push(item.getAsFile());
+      });
+    } else if (e.target?.files) {
+      [...e.target.files].forEach((file) => files.push(file));
+    }
+    return files;
+  }
+
+  async uploadImgToUnity(storageUrl, id, blobData, fileType) {
+    const uploadOptions = {
+      method: 'PUT',
+      headers: { 'Content-Type': fileType },
+      body: blobData,
+    };
+    const response = await fetch(storageUrl, uploadOptions);
+    return response.status === 200 ? id : '';
+  }
+
+  async scanImgForSafety(assetId) {
+    const assetData = { assetId, targetProduct: this.workflowCfg.productName };
+    const optionsBody = { body: JSON.stringify(assetData) };
+    const res = await this.serviceHandler.postCallToService(
+      this.psApiConfig.psEndPoint.acmpCheck,
+      optionsBody,
+      {},
+      false,
+    );
+    if (res.status === 429 || (res.status >= 500 && res.status < 600)) {
+      setTimeout(() => { this.scanImgForSafety(assetId); }, 1000);
+    }
+  }
+
+  async uploadAsset(file) {
+    try {
+      const resJson = await this.serviceHandler.postCallToService(
+        this.psApiConfig.psEndPoint.assetUpload,
+        {},
+        { errorToastEl: this.errorToastEl, errorType: '.icon-error-request' },
+      );
+      const { id, href } = resJson;
+      const assetId = await this.uploadImgToUnity(href, id, file, file.type);
+      this.scanImgForSafety(assetId);
+      return assetId;
+    } catch (e) {
+      await this.transitionScreen.showSplashScreen();
+      this.serviceHandler.showErrorToast({ errorToastEl: this.errorToastEl, errorType: '.icon-error-request' });
+      throw e;
+    }
+  }
+
+  async createErrorToast() {
```

Comment:
Should a try catch be added for these fetch requests?
```suggestion
  async createErrorToast() {
  try {
  // All code in this method
  } catch (error) {
    console.error('Failed to create error toast:', error);
  }
```